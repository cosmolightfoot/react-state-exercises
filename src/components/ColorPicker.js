import React, { PureComponent } from 'react';
import styles from './ColorPicker.css';

export default class ColorPicker extends PureComponent {
  state = {
    color: ''
  }

  setColor(color) {
    this.setState({ color });
    console.log('COLOR TEST', color);
  }

  render() {
    const { color } = this.state;

    return (
      <section className={styles.ColorPicker}>
        <button className={styles.red} onClick={() => this.setColor('red')}>Red</button> 
        <button className={styles.yellow} onClick={() => this.setColor('yellow')}>Yellow</button> 
        <button className={styles.blue} onClick={() => this.setColor('blue')}>Blue</button> 
        <div className={styles[color]}></div>
      </section>
    );
  }
}
