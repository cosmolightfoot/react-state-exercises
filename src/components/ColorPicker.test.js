import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker.js';

describe('ColorPicker tests', () => {
  it('renders ColorPicker', () => {
    const wrapper = shallow(<ColorPicker />);
    expect(wrapper).toMatchSnapshot();
  });

  it('changes color to red on button press', () => {
    const wrapper = shallow(<ColorPicker />);
    wrapper.find('button').at(0).simulate('click');

    expect(wrapper.state('color')).toEqual('red');
  });

  it('changes color to yellow on button press', () => {
    const wrapper = shallow(<ColorPicker />);
    wrapper.find('button').at(1).simulate('click');

    expect(wrapper.state('color')).toEqual('yellow');
  });

  it('changes color to blue on button press', () => {
    const wrapper = shallow(<ColorPicker />);
    wrapper.find('button').at(2).simulate('click');

    expect(wrapper.state('color')).toEqual('blue');
  });

  it('changes div className to color on button press', done => {
    const wrapper = shallow(<ColorPicker />);
    wrapper.setState({ color: 'red' }, () => {
      expect(wrapper.find('div').at(0).hasClass('red')).toBeTruthy();
      console.log(wrapper);
      done();
    });
  });
});
