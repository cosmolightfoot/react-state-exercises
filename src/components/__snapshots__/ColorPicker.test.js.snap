// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker tests renders ColorPicker 1`] = `
<section
  className="ColorPicker"
>
  <button
    className="red"
    onClick={[Function]}
  >
    Red
  </button>
  <button
    className="yellow"
    onClick={[Function]}
  >
    Yellow
  </button>
  <button
    className="blue"
    onClick={[Function]}
  >
    Blue
  </button>
  <div
    className=""
  />
</section>
`;
